### PR TITLE
[PORT?] adds flavour text

### PR DESCRIPTION
**Pull request by ToasterBan**
## About The Pull Request
ports https://github.com/Donkstation/Donkstation/pull/35 , with changes to be made as the version found in that PR seems to be of much lower code quality than that of ours

## Why It's Good For The Game
flavour text = better rp = good 

## Changelog
:cl: Toasterban, Imevs
add: Flavour Text!
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
Some real plannedemic shit.

**Comment by Crossedfall** [comment]
I've said it before, I'll say it again. Flavor text does not lead to better RP. This is a terrible addition. 

Forgot my key 🔑

**Review by Kapu1178 — CHANGES_REQUESTED** [review]
Needs `html_encode()` run on it to prevent HTML code execution.

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -1350,6 +1356,15 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 					else
 						to_chat(user, "<font color='red'>Invalid name. Your name should be at least 2 and at most [MAX_NAME_LEN] characters long. It may only contain the characters A-Z, a-z, -, ' and .</font>")
 
+				if("flavor_text")
+					var/new_text = input(user, "Please enter new flavor text (appears when examining you):", "Character Preference") as text|null
+					if (isnull(new_text))
```
Comment (Archanial) [inline code/modules/client/preferences.dm:1361]: isnull or empty

File: code/modules/client/preferences.dm [hunk 2]
```diff
@@ -1350,6 +1356,15 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
+						return
+					if (length(new_text) > FLAVOR_CHAR_LIMIT)
+						alert("Your flavor text is too long. It must be no more than [FLAVOR_CHAR_LIMIT] characters long. The current text will be trimmed down to meet the limit.")
+						new_text = copytext(new_text, 1, FLAVOR_CHAR_LIMIT+1)
```
Comment (Archanial) [inline code/modules/client/preferences.dm:1365]: Why +1?

File: code/modules/mob/living/carbon/examine.dm [hunk 3]
```diff
@@ -7,6 +7,14 @@
 	var/t_is = p_are()
 
 	. = list("<span class='info'>*---------*\nThis is [icon2html(src, user)] \a <EM>[src]</EM>!")
+
+
+	try
+		.+= "[src.flavor_text]"
+	catch
+
+	. +=  "<span class='notice'>*---------*</span>"
```
Comment (Archanial) [inline code/modules/mob/living/carbon/examine.dm:16]: ???

File: code/modules/mob/living/carbon/human/examine.dm [hunk 4]
```diff
@@ -16,6 +16,13 @@
 
 	. = list("<span class='info'>*---------*\nThis is <EM>[!obscure_name ? name : "Unknown"]</EM>!")
 
+	try
+		. += "<i>[src.flavor_text]</i>"
+	catch
+
+
+	. +=  "<span class='info'>*---------*</span>"
```
Comment (Archanial) [inline code/modules/mob/living/carbon/human/examine.dm:24]: ?????????

**Comment by Archanial** [comment]
my 🔑

**Pull request closed without merging**